### DISCUSSION TITLE
New Docker image and helm charts for Kubernetes

### DISCUSSION THREAD

**Discussion by mms-gianni** [haraka/Haraka#3171] [Show and tell]
Just found this project today on my way to find an SMTP Service for my other project [Kubero](https://github.com/kubero-dev/kubero). And I fell in love with it. 

I needed a simple to configure, feature full SMTP Service to run in my Kubernetes cluster. And Haraka provided exactly what I was looking for. 

Unfortunately is the Dockerfile a bit outdated. And I prefer alpine images due to their lower size, Memory footprint and vulnerabilities.

So I made a new Docker image. It is configurable by ENV vars (just implemented 10% of configurations) . This is necessary since I will run it in a Kubernetes cluster. 

I added also a helm chart to deploy it on Kubernetes. 

The trivy scan from today, which is quite good: 
```
ghcr.io/kubero-dev/haraka:latest (alpine 3.17.3)
Total: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)

Node.js (node-pkg)
Total: 9 (UNKNOWN: 0, LOW: 0, MEDIUM: 4, HIGH: 3, CRITICAL: 2)
```

I'm still working on the configuration part. And I'm not sure if I'll be able to make it 100% configurable, since I'm not a Mailadministrator and it already does what I needed. 

```
kubectl get all
NAME                          READY   STATUS    RESTARTS   AGE
pod/haraka-5975bf8978-65lqh   1/1     Running   0          28m

NAME             TYPE        CLUSTER-IP    EXTERNAL-IP   PORT(S)    AGE
service/haraka   ClusterIP   10.96.26.64   <none>        8025/TCP   49m

NAME                     READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/haraka   1/1     1            1           49m

NAME                                DESIRED   CURRENT   READY   AGE
replicaset.apps/haraka-5975bf8978   1         1         1       28m
```

So the Helm charts and Docker images can be found here: 
https://github.com/kubero-dev/haraka-docker

**Comment by baudehlo** [comment]
We may want to consider adding this to core -I feel like deployment solutions belong there. Would you be able to contribute it?On Apr 20, 2023, at 6:11 PM, Gianni C. ***@***.***> wrote:﻿
Just found this project today on my way to find an SMTP Service for my other project Kubero. And I fell in love with it.
I needed a simple to configure, feature full SMTP Service to run in my Kubernetes cluster. And Haraka provided exactly what I was looking for.
Unfortunately is the Dockerfile a bit outdated. And I prefer alpine images due to their lower size, Memory footprint and vulnerabilities.
So I made a new Docker image. It is configurable by ENV vars (just implemented 10% of configurations) . This is necessary since I will run it in a Kubernetes cluster.
I added also a helm chart to deploy it on Kubernetes.
The trivy scan from today, which is quite good:
ghcr.io/kubero-dev/haraka:latest (alpine 3.17.3)
Total: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)

Node.js (node-pkg)
Total: 9 (UNKNOWN: 0, LOW: 0, MEDIUM: 4, HIGH: 3, CRITICAL: 2)

I'm still working on the configuration part. And I'm not sure if I'll be able to make it 100% configurable, since I'm not a Mailadministrator and it already does what I needed.
kubectl get all
NAME                          READY   STATUS    RESTARTS   AGE
pod/haraka-5975bf8978-65lqh   1/1     Running   0          28m

NAME             TYPE        CLUSTER-IP    EXTERNAL-IP   PORT(S)    AGE
service/haraka   ClusterIP   10.96.26.64   <none>        8025/TCP   49m

NAME                     READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/haraka   1/1     1            1           49m

NAME                                DESIRED   CURRENT   READY   AGE
replicaset.apps/haraka-5975bf8978   1         1         1       28m

So the Helm charts and Docker images can be found here:
https://github.com/kubero-dev/haraka-docker

**Comment by baudehlo** [comment]
Also, awesome work. Loading everything from env vars is key these days. DJB was right about this too.On Apr 20, 2023, at 6:11 PM, Gianni C. ***@***.***> wrote:﻿
Just found this project today on my way to find an SMTP Service for my other project Kubero. And I fell in love with it.
I needed a simple to configure, feature full SMTP Service to run in my Kubernetes cluster. And Haraka provided exactly what I was looking for.
Unfortunately is the Dockerfile a bit outdated. And I prefer alpine images due to their lower size, Memory footprint and vulnerabilities.
So I made a new Docker image. It is configurable by ENV vars (just implemented 10% of configurations) . This is necessary since I will run it in a Kubernetes cluster.
I added also a helm chart to deploy it on Kubernetes.
The trivy scan from today, which is quite good:
ghcr.io/kubero-dev/haraka:latest (alpine 3.17.3)
Total: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)

Node.js (node-pkg)
Total: 9 (UNKNOWN: 0, LOW: 0, MEDIUM: 4, HIGH: 3, CRITICAL: 2)

I'm still working on the configuration part. And I'm not sure if I'll be able to make it 100% configurable, since I'm not a Mailadministrator and it already does what I needed.
kubectl get all
NAME                          READY   STATUS    RESTARTS   AGE
pod/haraka-5975bf8978-65lqh   1/1     Running   0          28m

NAME             TYPE        CLUSTER-IP    EXTERNAL-IP   PORT(S)    AGE
service/haraka   ClusterIP   10.96.26.64   <none>        8025/TCP   49m

NAME                     READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/haraka   1/1     1            1           49m

NAME                                DESIRED   CURRENT   READY   AGE
replicaset.apps/haraka-5975bf8978   1         1         1       28m

So the Helm charts and Docker images can be found here:
https://github.com/kubero-dev/haraka-docker
